src/babel.c

```diff
@@ -276,12 +276,22 @@ static gboolean babel_general_convert_from( VikTrwLayer *vt, BabelStatusFunc cb,
  */
 gboolean a_babel_convert_from_filter( VikTrwLayer *vt, const char *babelargs, const char *from, const char *babelfilters, BabelStatusFunc cb, gpointer user_data, gpointer not_used )
 {
-  int i,j;
   int fd_dst;
   gchar *name_dst = NULL;
   gboolean ret = FALSE;
   gchar *args[64];
 
+  /* directly read GPX file */
+  if ((babelargs != NULL && strcmp("-i gpx", babelargs) == 0) || a_file_check_ext(from, ".gpx")) {
+    FILE *f = g_fopen(from, "r");
+    if (f) {
+      ret = a_gpx_read_file(vt, f);
+      fclose(f);
+      return ret;
+    }
+  }
+
+  int i,j;
   if ((fd_dst = g_file_open_tmp("tmp-viking.XXXXXX", &name_dst, NULL)) >= 0) {
```

Comment:
This also seems irrelevant to the overall commit.

A further comment to explain why this useful (such as avoid running GPSBabel if possible).
I think it to be very rare that input arguments would only be "-i gpx".